Source/mpqapi.cpp

```diff
@@ -103,25 +103,21 @@ void __fastcall mpqapi_update_multi_creation_time(DWORD dwChar)
 
 BOOLEAN __fastcall mpqapi_reg_store_modification_time(char *pbData, int dwLen)
 {
-	int v2;          // ebx
-	char *v3;        // ebp
-	char *v4;        // edi
-	unsigned int v5; // esi
-	char *v6;        // ecx
+	char *pbCurrentData;
+	unsigned int i;
+	char *pbDataToXor;
 
-	v2 = dwLen;
-	v3 = pbData;
-	v4 = pbData;
+	pbCurrentData = pbData;
 	if ((unsigned int)dwLen >= 8) {

```

Comment:
dw* indicates that it's a DWORD and not an int. So probably change the type instead of casting.

Reply:
I tried it out with DWORD, it does not affect the binary exactness. I shall modify the PR accordingly.